src/rapids_dependency_file_generator/_rapids_dependency_file_generator.py

```diff
@@ -438,18 +483,50 @@ def make_dependency_files(
                 )
                 contents = make_dependency_file(
                     file_type=file_type,
-                    name=full_file_name,
+                    conda_env_name=os.path.splitext(full_file_name)[0],
+                    file_name=full_file_name,
                     config_file=parsed_config.path,
                     output_dir=output_dir,
-                    conda_channels=prepend_channels + parsed_config.channels,
+                    conda_channels=conda_channels,
                     dependencies=deduped_deps,
                     extras=file_config.extras,
                 )
 
                 if to_stdout:
-                    print(contents)
+                    if len(file_keys) == 1:
+                        print(contents)
+                    else:
+                        all_dependencies.update(deduped_deps)
                 else:
                     os.makedirs(output_dir, exist_ok=True)
                     file_path = os.path.join(output_dir, full_file_name)
                     with open(file_path, "w") as f:
                         f.write(contents)
+
+    # create one unified output from all the file_keys, and print it to stdout
+    if to_stdout and len(file_keys) > 1:
+        # convince mypy that 'output' is not None here
+        #
+        # 'output' is technically a set because of https://github.com/rapidsai/dependency-file-generator/pull/74,
+        # but since https://github.com/rapidsai/dependency-file-generator/pull/79 it's only ever one of the following:
+        #
+        #   - an exactly-1-item set (stdout=True, or when used by rapids-build-backend)
+        #   - 'None' (stdout=False)
+        #
+        err_msg = (
+            "Exactly 1 output type should be provided when asking rapids-dependency-file-generator to write to stdout. "
+            "If you see this, you've found a bug. Please report it at https://github.com/rapidsai/dependency-file-generator/issues."
+        )
+        assert output is not None, err_msg
+
+        contents = make_dependency_file(
+            file_type=output.pop(),
+            conda_env_name="rapids-dfg-combined",
```

Comment:
The `name: ` field in these generated conda environments is ignored in every workflow I'm aware of in RAPIDS... it's always overwritten by `-n` / `--name` passed through like this:

```shell
rapids-dependency-file-generator \
  --output conda \
  --file-key test_cpp \
  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch)" | tee "${ENV_YAML_DIR}/env.yaml"

mamba env create --yes -f "${ENV_YAML_DIR}/env.yaml" -n test
```

That's why I'm proposing a hard-coded mildly-informative name instead of taking on the complexity of other alternatives, like:

* allowing `conda_env_name=None` to be passed to `make_dependency_file()`
* trying to come up with some other representative environment name based on which `--file-key`s were passed

Reply:
Ok, I just pushed https://github.com/rapidsai/dependency-file-generator/pull/115/commits/df3848dd98d51c8aabbdad6adfcb6bfba75c1ea7 changing this to allow `None`.

> *I don't think conda environment files require a name?*

Correct.

```shell
cat > env.yaml <<EOF
channels:
- conda-forge
dependencies:
  - pandas
  - pip
  - pip:
    - scikit-learn
EOF

conda env create \
  --name delete-me \
  --file ./env.yaml
```

Works without issue

```
...
done
#
# To activate this environment, use
#
#     $ conda activate delete-me
#
# To deactivate an active environment, use
#
#     $ conda deactivat
```